### ISSUE TITLE
Meta-issue: Update all code to use optional new/const

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#18604]
Tracking issue for updating all our Flutter example code to use optional new/const (now that we are unblocked by the needed lint https://github.com/dart-lang/sdk/issues/32602 being available):

- [ ] Flutter.io homepage
- [x] `flutter/flutter/examples`
- [x] `flutter/plugins/examples`
- [x] `flutter/packages/examples`

**Comment by mit-mit** [comment]
Thanks, I reactivated https://github.com/dart-lang/sdk/issues/32596

**Comment by pq** [comment]
@a14n : once the new linter makes it's way into a vended SDK will we be good to make these changes?

**Comment by a14n** [comment]
I tried to remove unnecessary new/const from flutter codebase and faced a new issue in analyzer dart-lang/sdk#33647

**Comment by a14n** [comment]
Optional const landed in #19342

**Comment by pq** [comment]
@a14n: this is fantastic.  Remind me the plan for `unnecessary_new`?

**Comment by mit-mit** [comment]
@a14n do we have the no new lint?

**Comment by a14n** [comment]
`unnecessary_const` landed with #20103 and flutter/plugins#682.
`unnecessary_new` is waiting for reviews with #20138 and flutter/plugins#701.

**Comment by mit-mit** [comment]
Marking flutter/packages done as https://github.com/flutter/packages/pull/4 was completed

**Comment by mit-mit** [comment]
Assigning to self as only item left is updating images on flutter.io

**Comment by a14n** [comment]
Actually there's still flutter/plugins#818

**Comment by mit-mit** [comment]
This should be done now
